src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -149,6 +150,25 @@ public void ResetCompilerGenerated(bool newCompilerGenerated)
             }
         }
 
+        public bool IsSuppressed
+        {
+            get
+            {
+                return (_attributes & BoundNodeAttributes.IsSuppressed) != 0;
+            }
+            internal set
+            {
+
+                if (value)
+                {
+                    _attributes |= BoundNodeAttributes.IsSuppressed;
+                }
+                else
+                {
+                    Debug.Assert((_attributes & BoundNodeAttributes.IsSuppressed) == 0, "flag should not be reset");
```

Comment:
>Debug.Assert((_attributes & BoundNodeAttributes.IsSuppressed) == 0, "flag should not be reset"); [](start = 20, length = 96)

Should we assert this always (before the `if`), to ensure the bit is not set twice?